src/equations/compressible_euler_quasi_1d.jl

```diff
@@ -0,0 +1,368 @@
+    # Eleuterio F. Toro (2009)
+    # Riemann Solvers and Numerical Methods for Fluid Dynamics: A Practical Introduction
+    # [DOI: 10.1007/b79761](https://doi.org/10.1007/b79761)
+    if v_normal <= 0.0
+        sound_speed = sqrt(equations.gamma * p_local / rho_local) # local sound speed
+        p_star = p_local *
+                 (1 + 0.5 * (equations.gamma - 1) * v_normal / sound_speed)^(2 *
+                                                                             equations.gamma *
+                                                                             equations.inv_gamma_minus_one)
+    else # v_normal > 0.0
+        A = 2 / ((equations.gamma + 1) * rho_local)
+        B = p_local * (equations.gamma - 1) / (equations.gamma + 1)
+        p_star = p_local +
+                 0.5 * v_normal / A *
+                 (v_normal + sqrt(v_normal^2 + 4 * A * (p_local + B)))
+    end
+
+    # For the slip wall we directly set the flux as the normal velocity is zero
+    return SVector(zero(eltype(u_inner)),
+                   p_star,
+                   zero(eltype(u_inner)),
+                   zero(eltype(u_inner)))
+end
+
+# Calculate 1D flux for a single point
+@inline function Trixi.flux(u, orientation::Integer,
+                            equations::CompressibleEulerEquationsQuasi1D)
+    a_rho, a_rho_v1, a_e, a = u
+    rho, v1, p, a = cons2prim(u, equations)
+    e = a_e / a
+
+    # Ignore orientation since it is always "1" in 1D
+    f1 = a_rho_v1
+    f2 = a_rho_v1 * v1
+    f3 = a * v1 * (e + p)
+
+    return SVector(f1, f2, f3, zero(eltype(u)))
+end
+
+"""
+@inline function flux_nonconservative_chan_etal(u_ll, u_rr, orientation::Integer,
+                                                equations::CompressibleEulerEquationsQuasi1D)
+
+Non-symmetric two-point volume flux discretizing the nonconservative (source) term
+that contains the gradient of the pressure  [`CompressibleEulerEquationsQuasi1D`](@ref) 
+and the nozzle width.
+
+Further details are available in the paper:
+- Jesse Chan, Khemraj Shukla, Xinhui Wu, Ruofeng Liu, Prani Nalluri (2023)
+    High order entropy stable schemes for the quasi-one-dimensional
+    shallow water and compressible Euler equations
+    [DOI: 10.48550/arXiv.2307.12089](https://doi.org/10.48550/arXiv.2307.12089)    
+"""
+@inline function flux_nonconservative_chan_etal(u_ll, u_rr, orientation::Integer,
+                                                equations::CompressibleEulerEquationsQuasi1D)
+    #Variables
+    _, _, p_ll, a_ll = cons2prim(u_ll, equations)
+    _, _, p_rr, _ = cons2prim(u_rr, equations)
+
+    p_avg = 0.5 * (p_ll + p_rr)
+
+    z = zero(eltype(u_ll))
+
+    return SVector(z, 2 * a_ll * p_avg, z, z)
+end
+
+"""
+@inline function flux_chan_etal(u_ll, u_rr, orientation::Integer,
+                                           equations::CompressibleEulerEquationsQuasi1D)
+
+Total energy conservative (mathematical entropy for quasi 1D compressible Euler equations) split form.
```

Comment:
```suggestion
Conservative (symmetric) part of the entropy conservative flux for quasi 1D compressible Euler equations split form.
```
The entropy isn't equivalent to the total energy for the compressible Euler equations.